src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
+    private final Schema.SObjectField specificField;
+    private final QueryOperator specificOperator;
+    private final List<Object> specificValues;
+
+    public static QueryPredicate EQUALS(Schema.SObjectField field, Object value) {
```

Comment:
@jongpie the reason the constructors are private is because you can pass EITHER QueryPredicate.Equals(Lead.Id, expectedLead.Id) OR QueryPredicate.EQUALS(Lead.Id, new List<Id>{expectedLead.Id}) - with the point being that if you were to replace your second variable with ANOTHER list of Id's, the combination of predicate and QueryArgumentFormatter will work together in blissful harmony to decide whether or not to use = 'id here' or: IN (List of comma separated Id's here)